src/com/dabomstew/pkrandom/constants/Gen4Constants.java

```diff
@@ -150,18 +154,166 @@
+        map.put(dive, Arrays.asList(powerHerb));
+        map.put(fly, Arrays.asList(powerHerb));
+        map.put(razorWind, Arrays.asList(powerHerb));
+        map.put(skullBash, Arrays.asList(powerHerb));
+        map.put(skyAttack, Arrays.asList(powerHerb));
+        map.put(solarBeam, Arrays.asList(powerHerb));
+
+        map.put(fling, Arrays.asList(toxicOrb, flameOrb, ironBall));
+
+        map.put(trick, Arrays.asList(toxicOrb, flameOrb));
+        map.put(switcheroo, Arrays.asList(toxicOrb, flameOrb));
+
+        map.put(facade, Arrays.asList(toxicOrb, flameOrb));
+
+        map.put(psychoShift, Arrays.asList(toxicOrb, flameOrb));
+
+        map.put(lightScreen, Arrays.asList(lightClay));
+        map.put(reflect, Arrays.asList(lightClay));
+
+        map.put(hail, Arrays.asList(icyRock));
+
+        map.put(sandstorm, Arrays.asList(smoothRock));
 
+        map.put(sunnyDay, Arrays.asList(heatRock));
+
+        map.put(rainDance, Arrays.asList(dampRock));
+
+        map.put(bind, Arrays.asList(gripClaw));
+        map.put(clamp, Arrays.asList(gripClaw));
+        map.put(fireSpin, Arrays.asList(gripClaw));
+        map.put(magmaStorm, Arrays.asList(gripClaw));
+        map.put(outrage, Arrays.asList(gripClaw));
+        map.put(sandTomb, Arrays.asList(gripClaw));
+        map.put(uproar, Arrays.asList(gripClaw));
+        map.put(whirlpool, Arrays.asList(gripClaw));
+        map.put(wrap, Arrays.asList(gripClaw));
+
+        map.put(absorb, Arrays.asList(bigRoot));
+        map.put(aquaRing, Arrays.asList(bigRoot));
+        map.put(drainPunch, Arrays.asList(bigRoot));
+        map.put(dreamEater, Arrays.asList(bigRoot));
+        map.put(gigaDrain, Arrays.asList(bigRoot));
+        map.put(ingrain, Arrays.asList(bigRoot));
+        map.put(leechLife, Arrays.asList(bigRoot));
+        map.put(leechSeed, Arrays.asList(bigRoot));
+        map.put(megaDrain, Arrays.asList(bigRoot));
+
+        return Collections.unmodifiableMap(map);
+    }
+
+    public static final Map<Type, Integer> weaknessReducingBerries = initializeWeaknessReducingBerries();
+
+    private static Map<Type, Integer> initializeWeaknessReducingBerries() {
+        Map<Type, Integer> map = new HashMap<>();
+        map.put(Type.FIRE, occaBerry);
+        map.put(Type.WATER, passhoBerry);
+        map.put(Type.ELECTRIC, wacanBerry);
+        map.put(Type.GRASS, rindoBerry);
+        map.put(Type.ICE, yacheBerry);
+        map.put(Type.FIGHTING, chopleBerry);
+        map.put(Type.POISON, kebiaBerry);
+        map.put(Type.GROUND, shucaBerry);
+        map.put(Type.FLYING, cobaBerry);
+        map.put(Type.PSYCHIC, payapaBerry);
+        map.put(Type.BUG, tangaBerry);
+        map.put(Type.ROCK, chartiBerry);
+        map.put(Type.GHOST, kasibBerry);
+        map.put(Type.DRAGON, habanBerry);
+        map.put(Type.DARK, colburBerry);
+        map.put(Type.STEEL, babiriBerry);
+        // Note: Nothing is weak to normal, so the chilanBerry is "general purpose"
```

Comment:
Should the Chilan Berry be "General Purpose"? It's useless on Ghost types and mediocre on types that resist Normal. I think it warrants having a special case, where it's only put on Pokemon that take neutral damage from Normal.

Reply:
Done.